src/common/meta/src/rpc/router.rs

```diff
@@ -115,6 +118,50 @@ pub fn convert_to_region_map(region_routes: &[RegionRoute]) -> HashMap<u32, &Pee
         .collect::<HashMap<_, _>>()
 }
 
+/// Returns the HashMap<[RegionNumber], BTreeMap<DatanodeId, &[Peer]>>.
+pub fn convert_to_region_followers_map(
+    region_routes: &[RegionRoute],
+) -> HashMap<u32, BTreeMap<u64, &Peer>> {
+    region_routes
+        .iter()
+        .map(|x| {
+            (
+                x.region.id.region_number(),
+                x.follower_peers
+                    .iter()
+                    .map(|peer| (peer.id, peer))
+                    .collect::<BTreeMap<u64, &Peer>>(),
+            )
+        })
+        .collect::<HashMap<_, _>>()
+}
+
+/// Returns the HashMap<[RegionNumber], HashSet<DatanodeId>>.
+pub fn convert_to_region_peers_map(region_routes: &[RegionRoute]) -> HashMap<u32, HashSet<u64>> {
+    let mut set = region_routes
+        .iter()
+        .map(|x| {
+            (
+                x.region.id.region_number(),
+                x.follower_peers
+                    .iter()
```

Comment:
Is it possible to write it like this:

```suggestion
                    .iter().chain(std::iter::once(&route.leader_peer))
```

So you don't need to iter region_routes again below.